packages/entity-database-adapter-knex/src/PostgresEntityQueryContextProvider.ts

```diff
@@ -18,4 +18,10 @@ export default class PostgresEntityQueryContextProvider extends EntityQueryConte
   ) => Promise<T> {
     return (transactionScope) => this.knexInstance.transaction(transactionScope);
   }
+
+  protected createNestedTransactionRunner<T>(
+    outerQueryInterface: any
+  ): (transactionScope: (queryInterface: any) => Promise<T>) => Promise<T> {
+    return (transactionScope) => (outerQueryInterface as Knex).transaction(transactionScope);
```

Comment:
IIRC Postgres doesn't support nested BEGIN + COMMIT/ROLLBACKs. It uses SAVEPOINTS but I'm not sure Knex has support for that (I looked through the Knex source a little).